internal/provider/data_source_record.go

```diff
@@ -0,0 +1,80 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceCloudflareRecord() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceCloudflareRecordRead,
```

Comment:
need to add `Description` to this so that it will automatically fill out the docs.